packages/recoil-sync/RecoilSync_URL.js

```diff
@@ -55,6 +55,7 @@ function parseURL(
   href: string,
   loc: LocationOption,
   deserialize: string => mixed,
+  isQueryString: boolean,
```

Comment:
We should really try to support this without leaking the implementation through the abstraction.  Seems like this is looking to avoid wrapping to/from the `ItemSnapshot` Map, can you handle that distinction in `RecoilURLSyncQueryString`?  Is the issue here because `query-string` is trying to handle the complete query parameter string while `queryParams` mode handles individual parameters?  There may be a concern because part of the intention of `queryParams` mode is to be able to co-exist with other query parameters that the URL may have which need to pass through.  (I'll update the unit tests to cover that as well for JSON). Given the need to support arrays via multiple entries, I see where the existing `queryParams` loop is problematic..   Trying to think of a cleaner solution...

Reply:
Update the JSON tests as discussed in #2094

Reply:
Thanks for the review. 
I also a bit concerned about this. I'll think of the better solution.
We can still use `query-string`, right?

Reply:
I think it should be ok.. It uses the MIT license.  It does add some bloat, of course..

Trying to think if there's some solution leveraging the `pick()` method..

Reply:
> We should really try to support this without leaking the implementation through the abstraction.

@drarmstr I tried some ways without touching the abstraction, but It seems difficult to implement this feature without editing the abstraction because `queryString.stringify()` generates entire query params, however, `encodeURL()` uses [searchParams.toString()](https://github.com/facebookexperimental/Recoil/blob/main/packages/recoil-sync/RecoilSync_URL.js#L121). These are not compatible. 
Ofc, if we use `replace()`, I think we can implement it, but I don't think it's a good idea.

> Trying to think if there's some solution leveraging the pick() method..

If I understand it correctly, [pick()](https://github.com/sindresorhus/query-string#pickurl-keys-options) wouldn't be helpful in this case. 

Do you have any good idea to implement this feature without editing the abstraction?

If we can't find a good way, I don't mind closing this PR.